### ISSUE TITLE
Add negation operation in tags

### ISSUE THREAD

**Issue by lordofthejars** [asciidoctor/asciidoctor#1323]
As mention in the documentation if I have next document: 

```
// tag::snippets[] 
// tag::snippet-a[] 
snippet a 
// end::snippet-a[] 

// tag::snippet-b[] 
snippet b 
// end::snippet-b[] 
// end::snippets[]
```

And next include macro: 

```
include::file-with-snippets.adoc[tag=snippets]
```

Then the output is: 

snippet a 

snippet b

But I would like to be able to do something like: 

```
include::file-with-snippets.adoc[tag=snippets, !snippet-b]
```

Which would render to: 

snippets-a

Of course there is a workaround which is delimited all the sections you need but it is a tedious job and not always possible. I can try to implement it in asciidoctor if you agree it is a good addition.

**Comment by DavidGamba** [comment]
I think that syntax would be inconsistent with the lines syntax. It should be more like:

    include::file-with-snippets.adoc[tag="snippets,!snippet-b"]

or

    include::file-with-snippets.adoc[tag=snippets;!snippet-b]

Enclose in quotes when using comma or use semi colon otherwise.

Having said that, it would be inconsistent if we allow multiple entries in tags when using negation but don't allow this:

    include::file-with-snippets.adoc[tag=snippets-a;snippet-b]

To have multiple text snippets in the same include entry.

**Comment by lordofthejars** [comment]
> Enclose in quotes when using comma or use semi colon otherwise.
Yes of course this was my intention

**Comment by mojavelinux** [comment]
> Enclose in quotes when using comma or use semi colon otherwise.

Absolutely.

> To have multiple text snippets in the same include entry.

I'm pretty certain that's already possible. Am I wrong?

**Comment by DavidGamba** [comment]
> I'm pretty certain that's already possible. Am I wrong?

I tried just last night and wasn't able to get it to work with asciidoctor 1.5.2.

**Comment by lordofthejars** [comment]
I though the same as you but currently I am using in my book using:

```
include::{sourcedirchapter2}/{test-sources}/{service-directory}/{test-class}.java[tags="main,empty", indent=0]
```

**Comment by mojavelinux** [comment]
Aha. The attribute name has to be tags, not tag. The name tags works with
one or more tags. The name tag only allows one tag.

**Comment by DavidGamba** [comment]
> The attribute name has to be tags, not tag. The name tags works with
one or more tags. The name tag only allows one tag.

Thanks :-) for clarifying. I see that the docs on this were updated just yesterday too.

**Comment by mojavelinux** [comment]
I think #1516 is a duplicate of this issue.

**Comment by mojavelinux** [comment]
This was resolved in 1.5.6. See linked issue.
